tests/drivers/clock_control/stm32_clock_configuration/stm32h7_devices/src/test_stm32_clock_configuration.c

```diff
@@ -60,27 +60,48 @@ ZTEST(stm32h7_devices_clocks, test_spi_clk_config)
 		zassert_true((r == 0), "Could not enable SPI domain_clk");
 		TC_PRINT("SPI1 domain_clk on\n");
 
-		/* Test domain_clk is configured as device's source clock */
-		spi1_dt_domain_clk = COND_CODE_1(DT_CLOCKS_HAS_NAME(DT_NODELABEL(spi1), kernel),
-						  (DT_CLOCKS_CELL_BY_NAME(DT_NODELABEL(spi1),
-									  kernel, bus)),
-						  (DT_NO_CLOCK));
 		spi1_actual_domain_clk = __HAL_RCC_GET_SPI1_SOURCE();
 
-		if (spi1_dt_domain_clk == STM32_SRC_PLL1_Q) {
+		if (pclken[1].bus == STM32_SRC_PLL1_Q) {
 			zassert_equal(spi1_actual_domain_clk, RCC_SPI123CLKSOURCE_PLL,
-					"Expected SPI src: PLLQ (%d). Actual SPI src: %d",
+					"Expected SPI src: PLL1 Q (0x%x). Actual SPI src: 0x%x",
 					spi1_actual_domain_clk, RCC_SPI123CLKSOURCE_PLL);
-		} else if (spi1_dt_domain_clk == STM32_SRC_PLL3_P) {
+		} else if (pclken[1].bus == STM32_SRC_PLL3_P) {
```

Comment:
SPI123SEL[]  == 001 is also valid : **pll2_p_ck** selected as SPI/I2S1,2 and 3 kernel clock

Reply:
Agreed, but not proposed in tests configuration today.
I don't think the test configuration has to be exhaustive.